x-pack/platform/plugins/shared/lens/common/expressions/map_to_columns/map_to_columns_fn_textbased.ts

```diff
@@ -21,17 +21,39 @@ export const mapToOriginalColumnsTextBased: MapToColumnsExpressionFunction['fn']
       for (const id in row) {
         if (id in idMap) {
           for (const cachedEntry of idMap[id]) {
-            mappedRow[cachedEntry.id] = row[id]; // <= I wrote idMap rather than mappedRow
+            mappedRow[cachedEntry.id] = row[id];
+          }
+        } else {
+          const col = data.columns.find((c) => c.id === id);
+          if (col?.variable) {
+            const originalColumn = Object.values(idMap).find((idMapCol) => {
+              return idMapCol.some((c) => c.variable === col.variable);
+            });
+            if (originalColumn) {
+              for (const cachedEntry of originalColumn) {
+                mappedRow[cachedEntry.id] = row[id];
+              }
+            }
           }
         }
       }
 
       return mappedRow;
     }),
     columns: data.columns.flatMap((column) => {
-      if (!(column.id in idMap)) {
+      if (!(column.id in idMap) && !column.variable) {
         return [];
       }
+      if (column.variable) {
+        const originalColumn = Object.values(idMap).find((idMapCol) => {
```

Comment:
nit: the `idMapEntries` can be reused here to avoid to re-allocate memory for each column. Hopefully `|columns| << |rows|` so not super important in this case.

Reply:
Done https://github.com/elastic/kibana/pull/202875/commits/e69f8218c786f1b0288058d761013f3e10708a40